### ISSUE TITLE
is_symplectic(): broken batch handling - returns array instead of bool, missing omega broadcasting for batched inputs

### ISSUE THREAD

**Issue by Elshayib** [pnnl/hybridlane#85]
### Issue
In `src/hybridlane/math/symplectic.py`, the `is_symplectic()` function has two bugs:

1. **Incorrect batch reduction** - For batched inputs, it returns an array of booleans instead of a single boolean
2. **Missing omega broadcasting for batched inputs** - The symplectic form `omega` is 2D but batched `Sc` is 3D, causing incorrect matrix multiplication

### Location
`src/hybridlane/math/symplectic.py` lines 150-180 (approximately)

#### Bug 1: Incorrect batch reduction
```python
def is_symplectic(S: TensorLike, rtol=1e-5, atol=1e-8) -> bool:
    # ...
    batch_size = math.get_batch_size(S, (n, n), n**2)
    if batch_size is not None:
        transpose_axes = (0, 2, 1)
    else:
        transpose_axes = (1, 0)

    S = math.asarray(S)
    omega = symplectic_form(n_modes, like=math.get_interface(S))
    Sc = S[..., 1:, 1:] if has_constant else S
    residual = math.transpose(Sc, axes=transpose_axes) @ omega @ Sc - omega
    return math.all(math.isclose(residual, 0, rtol=rtol, atol=atol), axis=(-1, -2))
```

**Problem:** For batched input of shape `(batch, 2n, 2n)`, the `axis=(-1, -2)` reduces only the matrix dimensions, leaving a result of shape `(batch,)`. The function signature says `-> bool` but returns an array for batched inputs.

**Expected:** Should reduce over all axes including batch to return a single boolean, or document that batched inputs return batched booleans.

#### Bug 2: Missing omega broadcasting for batched matrix multiplication
```python
omega = symplectic_form(n_modes, like=math.get_interface(S))
# omega shape: (2n_modes, 2n_modes) = (2n, 2n) - 2D
Sc = S[..., 1:, 1:] if has_constant else S
# Sc shape for batched: (batch, 2n, 2n) - 3D
residual = math.transpose(Sc, axes=transpose_axes) @ omega @ Sc - omega
```

**Problem:** `omega` is 2D `(2n, 2n)` but `Sc` is 3D `(batch, 2n, 2n)`. In standard array libraries (NumPy, JAX, PyTorch, TensorFlow), `(batch, 2n, 2n) @ (2n, 2n)` does NOT correctly broadcast - it either errors or produces incorrect results. The `omega` needs to be expanded to `(1, 2n, 2n)` or `(batch, 2n, 2n)` for proper broadcasting.

### Reproduction
```python
import numpy as np
from hybridlane.math.symplectic import is_symplectic, symplectic_form

# Test 1: Batched identity matrices (should all be symplectic)
S_batch = np.stack([np.eye(3), np.eye(3), np.eye(3)])  # shape (3, 3, 3)
result = is_symplectic(S_batch)
print(f"Batched result: {result}, shape: {result.shape}")
# Expected: True (or array([True, True, True]) if batched return is intended)
# Actual: array([True, True, True]) - returns array not bool

# Test 2: Non-symplectic batch
S_bad = np.stack([np.array([[2, 1, 0], [0, 0, 0], [0, 0, 0]]), 
                  np.eye(3)])
result = is_symplectic(S_bad)
# With bug 2, the omega broadcasting issue may give incorrect results
```

### Suggested Fix

```python
def is_symplectic(S: TensorLike, rtol=1e-5, atol=1e-8) -> bool:
    if math.ndim(S) < 2 or ((shape := math.shape(S)) and shape[-1] != shape[-2]):
        raise ValueError(f"Input must be a square matrix, got shape {shape}")

    n = shape[-1]
    n_modes = n // 2
    has_constant = bool(n % 2)

    batch_size = math.get_batch_size(S, (n, n), n**2)
    if batch_size is not None:
        transpose_axes = (0, 2, 1)
    else:
        transpose_axes = (1, 0)

    S = math.asarray(S)
    omega = symplectic_form(n_modes, like=math.get_interface(S))
    Sc = S[..., 1:, 1:] if has_constant else S
    
    # Fix Bug 2: Expand omega for proper broadcasting with batched inputs
    if batch_size is not None:
        # omega shape (2n, 2n) -> (1, 2n, 2n) for broadcasting
        omega = math.expand_dims(omega, 0)
    
    residual = math.transpose(Sc, axes=transpose_axes) @ omega @ Sc - omega
    
    # Fix Bug 1: Reduce over ALL axes including batch to return single bool
    return math.all(math.isclose(residual, 0, rtol=rtol, atol=atol))
```

### Impact
- **Medium-High**: Batched symplecticity checks return arrays instead of bools, breaking code that expects `bool` return type
- **Medium**: Batched checks may give mathematically incorrect results due to missing omega broadcasting
- The function is part of the public API (`__all__` includes `is_symplectic`)

**Comment by jfurches** [comment]
Hi, thanks for submitting a bug report!

**Bug 1**: Regarding bug 1, batching is the intended behavior, and calling `is_symplectic` with a matrix of shape `(B, n, n)` should treat it as a stack of `B` matrices and return a 1D array of shape `(B,)`. The reduction "fix" is unnecessary. This behavior is lightly mentioned in the docstring of the method, so it could perhaps be improved and the type signature updated to `bool | TensorLike`.

**Bug 2**: Could you provide a minimum working example showing that the matmul fails? I have this test evaluating batching behavior:

https://github.com/pnnl/hybridlane/blob/f84d60f0b59421c393908cef3be113d79842857f/test/math/test_symplectic.py#L109-L122

Furthermore, the [NumPy docs on matmul](https://numpy.org/doc/2.1/reference/generated/numpy.matmul.html) state that

> If either argument is N-D, N > 2, it is treated as a stack of matrices residing in the last two indexes and broadcast accordingly.
